game-core/src/main/java/games/strategy/engine/random/PropertiesDiceRoller.java

```diff
@@ -128,32 +93,21 @@ public String postRequest(
   @Override
   public int[] getDice(final String string, final int count)
       throws IOException, InvocationTargetException {
-    final String errorStartString = props.getProperty("error.start");
-    final String errorEndString = props.getProperty("error.end");
-    // if the error strings are defined
-    if (errorStartString != null
-        && errorStartString.length() > 0
-        && errorEndString != null
-        && errorEndString.length() > 0) {
-      final int startIndex = string.indexOf(errorStartString);
-      if (startIndex >= 0) {
-        final int endIndex =
-            string.indexOf(errorEndString, (startIndex + errorStartString.length()));
-        if (endIndex > 0) {
-          final String error = string.substring(startIndex + errorStartString.length(), endIndex);
-          throw new InvocationTargetException(null, error);
-        }
+    final String errorStartString = "fatal error:";
+    final String errorEndString = "!";
+    final int errorStringStartIndex = string.indexOf(errorStartString);
+    if (errorStringStartIndex >= 0) {
+      final int endIndex =
+          string.indexOf(errorEndString, (errorStringStartIndex + errorStartString.length()));
+      if (endIndex > 0) {
+        final String error =
+            string.substring(errorStringStartIndex + errorStartString.length(), endIndex);
+        throw new InvocationTargetException(null, error);
       }
     }
-    final String rollStartString;
-    final String rollEndString;
-    if (count == 1) {
-      rollStartString = props.getProperty("roll.single.start");
-      rollEndString = props.getProperty("roll.single.end");
-    } else {
-      rollStartString = props.getProperty("roll.multiple.start");
-      rollEndString = props.getProperty("roll.multiple.end");
-    }
+
+    final String rollStartString = "your dice are:";
+    final String rollEndString = "<p>";
     int startIndex = string.indexOf(rollStartString);
     if (startIndex == -1) {
       throw new IOException("Could not find start index, text returned is:" + string);
```

Comment:
Same here with the preprocessing step:
```java
Pattern pattern = Pattern.compile("your dice are:(.*)<p>");
// Side Note: When using Pattern.compile("your dice are:(\s*\d+\s*(?:,\s*\d+\s*)*)<p>")
// you could probably avoid having the try-catch block below
Matcher matcher = pattern.matcher(string);
if (!matcher.find()) {
  throw new IOException("String doesn't contain dice information");
}

// Existing code...
// replace string.substring(startIndex, endIndex) with matcher.group(1) below
```

Reply:
Agree, looks right, though same response that unless/until we are also willing to test this, it's a risky change to make.

Stepping back, the content is being served as HTML from MARTI, we will be moving to a JSON. It may be more worthwhile to replace rather than improve this code with a JSON API.